receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -104,6 +105,19 @@ func (s *scraper) start(context.Context, component.Host) error {
 func (s *scraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	var errs scrapererror.ScrapeErrors
 
+	// If the boot time cache featuregate is disabled, this will refresh the
+	// cached boot time value for use in the current scrape. This functionally
+	// replicates the previous functionality in all but the most extreme
+	// cases of boot time changing in the middle of a scrape.
+	if !bootTimeCacheFeaturegate.IsEnabled() {
+		host.EnableBootTimeCache(false)
+		_, err := host.BootTimeWithContext(ctx)
+		if err != nil {
+			errs.AddPartial(1, fmt.Errorf(`retrieving boot time failed with error "%w", using cached boot time`, err))
+		}
+		host.EnableBootTimeCache(true)
```

Comment:
This line should go after the condition block, not inside, right? Otherwise the FG is a no-op

Reply:
The boot time cache being enabled happens in the hostmetrics when the receiver is created. This particular disabling and immediate re-enabling of the cache is just to refresh the cached value.

Reply:
Oh I missed that part 👍